goto-chg.el

```diff
@@ -331,7 +333,7 @@ discarded. See variable `undo-limit'."
             (when (not glc-seen-canary)
               (setq l (cdr l)))))
         (when glc-seen-canary
-          (while (and (< n new-probe-depth) (undo-tree-node-p l))
+          (while (and (< n new-probe-depth) (eval '(undo-tree-node-p l)))
```

Comment:
That's an interesting solution alright. I'll file it under "Legitimate uses of eval".